sql/core/src/main/scala/org/apache/spark/sql/jdbc/JdbcDialects.scala

```diff
@@ -633,13 +633,31 @@ abstract class JdbcDialect extends Serializable with Logging {
    * @param e The dialect specific exception.
    * @param errorClass The error class assigned in the case of an unclassified `e`
    * @param messageParameters The message parameters of `errorClass`
+   * @param description The error description
    * @return `AnalysisException` or its sub-class.
    */
   def classifyException(
       e: Throwable,
       errorClass: String,
-      messageParameters: Map[String, String]): AnalysisException = {
-    new AnalysisException(errorClass, messageParameters, cause = Some(e))
+      messageParameters: Map[String, String],
+      description: String): AnalysisException = {
+    classifyException(description, e)
```

Comment:
Instead of passing `description` around, shall we generate the description here using the error class name and the `tableName` in the parameters?

Reply:
This might be not fully compatible with existing JDBC dialects because values in `messageParameters` has been preprocessed already like quoting, and a JDBC dialect could use regexp for parsing parameters from `mesage`/`description`. Theoretically, it can break the regexps.